flw/src/main/java/org/motechproject/nms/flw/service/impl/ServiceUsageCapServiceImpl.java

```diff
@@ -0,0 +1,82 @@
+package org.motechproject.nms.flw.service.impl;
+
+import org.motechproject.mds.query.QueryExecution;
+import org.motechproject.mds.util.InstanceSecurityRestriction;
+import org.motechproject.nms.flw.domain.ServiceUsageCap;
+import org.motechproject.nms.flw.repository.ServiceUsageCapDataService;
+import org.motechproject.nms.flw.service.ServiceUsageCapService;
+import org.motechproject.nms.location.domain.State;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.jdo.Query;
+
+/**
+ * Created by rob on 4/15/15.
+ */
+@Service("serviceUsageCapService")
+public class ServiceUsageCapServiceImpl implements ServiceUsageCapService {
+    private ServiceUsageCapDataService serviceUsageCapDataService;
+
+    @Autowired
+    public ServiceUsageCapServiceImpl(ServiceUsageCapDataService serviceUsageCapDataService) {
+        this.serviceUsageCapDataService = serviceUsageCapDataService;
+    }
+
+    /*
+    The spec was a little unclear on which cap took precedence.
+    (Sara in mail 4/15/15: it should be possible to set both a national cap or a state cap. If you set a
+                           state cap for a service, then the national cap does not apply. If no state cap is set, then
+                           the national cap applies.
+     */
+    @Override
+    public ServiceUsageCap getServiceUsageCap(final State state, final org.motechproject.nms.flw.domain.Service service) {
```

Comment:
I think this class also import springs 'Service' so there was a conflict

On Mon, Apr 20, 2015 at 1:18 PM, Lauren Lavoie <notifications@github.com>
wrote:

> In
> flw/src/main/java/org/motechproject/nms/flw/service/impl/ServiceUsageCapServiceImpl.java
> <https://github.com/koshalt/mim/pull/46#discussion_r28724831>:
>
> > +public class ServiceUsageCapServiceImpl implements ServiceUsageCapService {
> > +    private ServiceUsageCapDataService serviceUsageCapDataService;
> > +
> > +    @Autowired
> > +    public ServiceUsageCapServiceImpl(ServiceUsageCapDataService serviceUsageCapDataService) {
> > +        this.serviceUsageCapDataService = serviceUsageCapDataService;
> > +    }
> > +
> > +    /*
> > +    The spec was a little unclear on which cap took precedence.
> > +    (Sara in mail 4/15/15: it should be possible to set both a national cap or a state cap. If you set a
> > +                           state cap for a service, then the national cap does not apply. If no state cap is set, then
> > +                           the national cap applies.
> > +     */
> > +    @Override
> > +    public ServiceUsageCap getServiceUsageCap(final State state, final org.motechproject.nms.flw.domain.Service service) {
>
> Why the fully qualified identifier here rather than importing
> org.motechproject.nms.flw.domain.Service?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/koshalt/mim/pull/46/files#r28724831>.
>